vsintegration/src/FSharp.Editor/LanguageService/LanguageService.fs

```diff
@@ -316,12 +317,28 @@ and
     override this.Initialize() =
         base.Initialize()
 
+        let workspaceChanged (args:WorkspaceChangeEventArgs) =
+            match args.Kind with
+            | WorkspaceChangeKind.ProjectAdded
+            | WorkspaceChangeKind.ProjectChanged
```

Comment:
The event tells you the affect projects in these cases, instead of needing to iterate all projects